manifests/agent.pp

```diff
@@ -0,0 +1,51 @@
+class ss_nessus_agent::agent inherits ::ss_nessus_agent {
+
+  if $ss_nessus_agent::http_proxy {
+      $proxy_environment = ["http_proxy=${ss_nessus_agent::http_proxy}", "https_proxy=${ss_nessus_agent::http_proxy}"]
+  } else {
+      $proxy_environment = []
+  }
+
+  exec { 'download':
+    command     => "curl -s -f ${ss_nessus_agent::download_url} -o /usr/src/nessusagent.deb",
+    path        => '/usr/bin:/usr/sbin:/bin',
+    onlyif      => "test ! -f /usr/src/nessusagent.deb",
+    environment => $proxy_environment,
+  }
+  -> package { 'nessusagent':
+    ensure   => installed,
+    provider => dpkg,
+    source   => '/usr/src/nessusagent.deb',
+  }
+
+  # The setup and configuration utilises the custom facts
+  if $params::nessus_agent_enabled == true {
+    if !$facts['ss_nessus_agent_connected'] {
+      exec { 'ss_nessus_agent_install_agent':
+        command => "sudo -u root dpkg -i /usr/src/nessusagent.deb",
+        path    => '/usr/bin:/usr/sbin:/bin',
+      }
+      -> exec { 'ss_nessus_agent_start_agent':
+        command => "sudo -u root systemctl start nessusagent",
+        path    => '/usr/bin:/usr/sbin:/bin',
+      }
+      -> exec { 'ss_nessus_agent_link_agent':
+        command => "/opt/nessus_agent/sbin/nessuscli agent link --key=${ss_nessus_agent::link_token} name=${$facts['environmentname']} --cloud --proxy-host=${ss_nessus_agent::http_proxy_host} --proxy-port=${ss_nessus_agent::http_proxy_port}",
```

Comment:
Shouldn't assume `$facts['environmentname']` will exist. Instead use a parameter on the module and have this information passed to it within CCL

Reply:
does `--proxy-host=${ss_nessus_agent::http_proxy_host} --proxy-port=${ss_nessus_agent::http_proxy_port}"` allow for no proxy information to be provided?

E.g. If proxy variables are empty, will this command still work?

On AWS we don't use proxies, so these variables will be blank/empty.

If it doesn't work with blank entries, then we may need some logic around if proxy information exists, then append these flags etc.